### PR TITLE
Simplifications

### PR DESCRIPTION
**Pull request by costastf**
Hi Fred,
I went over the uhttpd code since I am using it and have made some changes, which are mostly simplifications, semantic renames and restructures and some introspection lookups for http verbs and http codes that would help the extendability of the code. I hope you don't mind the range of the changes. Please do not hesitate to contact me for for anything and again thank you for this amazing module!

### CODE REVIEW

**Comment by fadushin** [comment]
I don't mind at all!  I will review this shortly.  I have been away from ESP development, and I will get back to it in short time.  Thank you for your contributions!  (Do the changes pass the tests?)

**Comment by costastf** [comment]
Hi Fred,

Sorry,I didn't notice any testing suite so I don't know.

Kind regards
Costas

File: micropython/uhttpd/http_api_handler.py [hunk 1]
```diff
@@ -107,20 +106,22 @@ def find_handler(self, components):
...
-                raise uhttpd.BadRequestException("Invalid query parameter: {}".format(qparam_component))
-            query_params[qparam[0]] = qparam[1]
-        return path_part, query_params
+        components = [component for component in query.split("&")
+                      if component.strip()]
+        for qparam_component in components:
+            try:
+                parameter, value = qparam_component.split("=")
+                if not parameter:
+                    raise ValueError
```
Comment (fadushin) [inline micropython/uhttpd/http_api_handler.py:125]: Not to keen on raising an exception and then catching it in the same clause.  Would prefer we raise an exception at that point with an even more detailed reason.

File: micropython/uhttpd/http_api_handler.py [hunk 2]
```diff
@@ -41,57 +41,56 @@ def handle_request(self, http_request):
...
-            elif verb == 'put':
-                response = handler.put(api_request)
-            elif verb == 'post':
-                response = handler.post(api_request)
-            elif verb == 'delete':
-                response = handler.delete(api_request)
-            else:
+            try:
+                response = getattr(handler, verb)(api_request)
+                # if verb == 'get':
```
Comment (fadushin) [inline micropython/uhttpd/http_api_handler.py:62]: Sounds good.  Let's remove the commented bits then.

File: micropython/uhttpd/http_file_handler.py [hunk 3]
```diff
@@ -211,23 +214,16 @@ def components(self, path):
...
     def create_response(code, content_type, length, body):
-        return {
-            'code': code,
-            'headers': {
-                'content-type': content_type,
-                'content-length': length
-            },
-            'body': body
-        }
+        return {'code': code,
```
Comment (fadushin) [inline micropython/uhttpd/http_file_handler.py:223]: Sorry, I prefer the indentation style I am using.  I realize it is non-standard, but it is regular, which why I prefer it.  (Consistency is better than ad-hoc)

**Review by fadushin — APPROVED** [review]

**Review by fadushin — APPROVED** [review]

**Comment by fadushin** [comment]
I had to revert this merge, because the tests were failing, and I don't really have time to look into the changes.  Perhaps make corrections to fix the tests, and then resubmit?  Sorry!

**Comment by costastf** [comment]
Hi Fred, yes, of course I will look into the tests and make sure that everything works and I will make a new PR as soon as I have time to work on it. Sorry for the trouble.

**Pull request merged**